### DISCUSSION TITLE
Flet app with sqlite3

### DISCUSSION THREAD

**Discussion by marcrr79907** [flet-dev/flet#3876] [Q&A]
### Question

Hello, I have an app in Flet that obtains information from a .sqlite3 database, the entire interface displays well with little data, the problem arises when reading information from a database with a larger amount of data. I have access to all the information without errors, but the interface is not displayed, I think it may be due to the volume of data, although it is a 600kb .sqlite3 file. Please if you have any idea on how to optimize the process tell me.

### Code sample

```python
# Get data example
    def get_houses(self):
        try:
            query = self.cursor.execute('SELECT * FROM tarifario_casacomercial')
            return list(query.fetchall())
        except sql.OperationalError:
            self.pg.go('/error')
# Show data
    def get_houses(self):

        house_list = Connect_DB(self.pg).get_houses()

        if house_list:
            for house in house_list:
                lines_names = []
                lines = Connect_DB(self.pg).get_lines_x_house(house[0])
                num_actions = Connect_DB(self.pg).get_number_services_x_house(house[0])

                for line in lines:
                    lines_names.append(line[1])

                house = House(house[1], house[0], lines_names, num_actions)

                self.object_list.append(
                    house
                )

        return self.object_list
```

### Error message

```bash
Only blank screen is shown, no errors.
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Give this idea a try: https://flet.dev/docs/cookbook/large-lists/#batch-updates

**Comment by eliezerke** [comment]
**Async await implementation**
use asyncio python module and make your functions utilize the async function while retrieving data from the database to ensure that needed data is only fetched when needed.
@marcrr79907 #3024 #3337

**Comment by hitzsf** [comment]
You can create a custom pagination control.
